test/functional/p2p_filter.py

```diff
@@ -135,14 +136,32 @@ def test_msg_mempool(self):
         self.log.info("Check that a node with bloom filters enabled services p2p mempool messages")
         filter_peer = P2PBloomFilter()
 
-        self.log.debug("Create a tx relevant to the peer before connecting")
-        txid = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=filter_peer.watch_script_pubkey, amount=9 * COIN)["txid"]
+        self.log.info("Create two tx before connecting, one relevant to the node another that is not")
+        rel_txid = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=filter_peer.watch_script_pubkey, amount=1 * COIN)["txid"]
+        irr_txid = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=getnewdestination()[1], amount=2 * COIN)["txid"]
+        irr_wtxid = self.nodes[0].getmempoolentry(irr_txid)["wtxid"]
 
-        self.log.debug("Send a mempool msg after connecting and check that the tx is received")
+        self.log.info("Send a mempool msg after connecting and check that the relevant tx is received")
         self.nodes[0].add_p2p_connection(filter_peer)
-        filter_peer.send_and_ping(filter_peer.watch_filter_init)
-        filter_peer.send_message(msg_mempool())
-        filter_peer.wait_for_tx(txid)
+        filter_peer.send_message(filter_peer.watch_filter_init)
+        filter_peer.send_and_ping(msg_mempool())
+        filter_peer.wait_for_tx(rel_txid)
+
+        self.log.info("Request the irrelevant transaction even though it has not being offered to us")
+        filter_peer.tx_received = False
+        filter_peer.send_and_ping(msg_getdata([CInv(t=MSG_WTX, h=int(irr_wtxid, 16))]))
+        self.log.info("We should get it anyway given it was on the mempool when the MEMPOOL message was sent")
+        assert filter_peer.tx_received
```

Comment:
instead of using `tx_received`, `wait_for_tx` will also assert that you're receiving the right tx:
```suggestion
        filter_peer.wait_for_tx(irr_txid)
```